tools/matRad_spotRemoval.m

```diff
@@ -0,0 +1,99 @@
+function [dij,stf] = matRad_spotRemoval(dij,w,stf,thres)
+% matRad postprosseing function accounting for
+%       minimum number of particles per spot
+%       minimum number of particles per iso-energy slice
+%
+% call
+%   resultGUI =  matRad_postprocessing(resultGUI, dij, pln, cst, stf)
+
+% input
+%   resultGUI   struct containing optimized fluence vector
+%   dij:        matRad dij struct
+%   pln:        matRad pln struct
+%   cst:        matRad cst struct
+%   stf:        matRad stf struct
+%
+% output
+%   resultGUI:  new w and doses in resultGUI
+%
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2021 the matRad development team.
+%
+% This file is part of the matRad project. It is subject to the license
+% terms in the LICENSE file found in the top-level directory of this
+% distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part
+% of the matRad project, including this file, may be copied, modified,
+% propagated, or distributed except according to the terms contained in the
+% LICENSE file.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+matRad_cfg = MatRad_Config.instance();
+
+if nargin < 3 && nargout > 1
+    matRad_cfg.dispError('stf is missing as input')
+end
+
+% set threshold for spot removal to 3% of the maximum weight.
+if ~exist('thres','var')
```

Comment:
so the thresshold is strictly relative?
It might also make sense to think about absolute values here, since accelerators usally have a hard lower limit on how much particles they output minimally.